pkg/controller/common/certificates/ca_secret.go

```diff
@@ -12,23 +12,36 @@ import (
 // ParseCustomCASecret checks that mandatory fields are present and returns a CA struct.
 // It does not check that the public key matches the private key.
 func ParseCustomCASecret(s corev1.Secret) (*CA, error) {
+	keyFileName := CAKeyFileName
```

Comment:
Should this function name (or at least godoc) indicate allowing legacy key naming? I'm not sure how one decides calling `ParseCustomCASecret` vs `parseCAFromSecret` (for example from `validateCustomCA`).

Reply:
Good point. I actually want to hide the `parseCAFromSecret` one as an internal function that just avoids duplication between the different parsing functions and meant `ParseCustomCASecret` as the actual API.